core/src/engine/EngineMultiRobot.cc

```diff
@@ -2093,6 +2091,53 @@ namespace jiminy
         return returnCode;
     }
 
+    vectorN_t EngineMultiRobot::computeAcceleration(std::shared_ptr<Robot> robot,
+                                                    vectorN_t const & q,
+                                                    vectorN_t const & v,
+                                                    vectorN_t const & u,
+                                                    forceVector_t const & fext)
+    {
+        if (robot->hasConstraint())
+        {
+            // Handle kinematic constraints.
+            matrixN_t J;
+            vectorN_t drift;
+            robot->computeConstraints(q, v, J, drift);
+
+            // Project external forces from cartesian space to joint space.
+            vectorN_t uTotal = u;
```

Comment:
Do not allocate memory but rather use buffer (for example using a struct ConstraintData_t if they are many)

Reply:
.